scipy/stats/_stats_py.py

```diff
@@ -8498,27 +8503,28 @@ def kruskal(*samples, nan_policy='propagate'):
     KruskalResult(statistic=7.0, pvalue=0.0301973834223185)
 
     """
-    samples = list(map(np.asarray, samples))
-
     num_groups = len(samples)
     if num_groups < 2:
         raise ValueError("Need at least two groups in stats.kruskal()")
 
-    n = np.asarray(list(map(len, samples)))
+    n = np.asarray([sample.shape[-1] for sample in samples])
+    totaln = np.sum(n)
```

Comment:
`n` is a 1-D array, so I left `axis` argument out to emphasize that `totaln` will be a scalar. Probably should go ahead and cast to `float` now; this will avoid possible mixed float-int issues with array API strict later.
```suggestion
    totaln = float(np.sum(n))
```